frontend/packages/console-shared/src/components/dashboard/utilization-card/prometheus-hook.ts

```diff
@@ -35,4 +37,41 @@ export const usePrometheusQuery: UsePrometheusQuery = (query, humanize) => {
   return results;
 };
 
+const customSelectorCreator = createSelectorCreator(defaultMemoize, shallowEqual);
+
+export const usePrometheusQueries: UsePrometheusQueries = (queries, metric) => {
+  const dispatch = useDispatch();
+  React.useEffect(() => {
+    queries.forEach((query) => dispatch(watchPrometheusQuery(query)));
+    return () => {
+      queries.forEach((query) => dispatch(stopWatchPrometheusQuery(query)));
+    };
+  }, [dispatch, queries]);
+
+  const selectors = React.useMemo(
+    () => queries.map((q) => ({ dashboards }) => dashboards.getIn([RESULTS_TYPE.PROMETHEUS, q])),
+    [queries],
+  );
+
+  const querySelector = React.useMemo(() => customSelectorCreator(selectors, (...data) => data), [
+    selectors,
+  ]);
+
+  const queryResults = useSelector<RootState, ImmutableMap<string, any>>(querySelector);
+
+  const results = React.useMemo<UsePrometheusQueriesResult>(() => {
+    if (_.isEmpty(queryResults?.[0])) {
```

Comment:
I think we should return `[null, null]` for not yet loaded queries. The current implementation would return something like `[[], null]`

Maybe like this ?
```javascript
if (_.isEmpty(queryResults)) {
  return queries.map(() => [null, null]) as UsePrometheusQueriesResult;
}

return queryResults.reduce((acc, curr) => {
  const data = curr.get('data');
  const value = data ? getInstantVectorStats(data, metric) : null;
  return [...acc, [value, curr.get('loadError')]];
}, []);
```

Reply:
as disucssed making changes to align with useWatchK8sResource `[data, loading, loadError][]`